### ISSUE TITLE
More features need?

### ISSUE THREAD

**Issue by EvgeniyMakhmudov** [fabiospampinato/vscode-todo-plus#10]
Previously i used SublimeText with plugin [PlainTasks](https://github.com/aziz/PlainTasks). Todo+ right now have no useful features, such add datetime stamp when task done/canceled, "archive" mode. Last feature really cool, it allow often not use scroll of screen to show all undone tasks, because all completed tasks you can send in "archive" by simple command or keys combination and keep you lists actual.

**Comment by fabiospampinato** [comment]
I'm skeptical about the usefulness of those features.

Timestamp: in which case is this really useful? I think most of the times any completed todo corresponds to a new commit, which already provides this information.

Archive: having the ability to auto-remove completed todos is interesting, but VS Code already provides a `Delete Line` command which basically does just that. Having todos archived in another file or moved to the bottom of the file seems useless -- having a task inside a project or sub-todo adds context to it, removing this information makes that todo not so useful anymore.

**Comment by EvgeniyMakhmudov** [comment]
1. You think not so wide as possible. Task is not equal commit. Task is some note for yourself to remember do anything, such make call to colleague to discuss some feature, check changelog in new version of packages and even private tasks like buy new stuff to home. So, that task generally may have no link to any timestamp in other place, so it may be useful, if necessary next time to remember when tasks was complete.
2. Keeping todo file with actual tasks, when completed moved to the bottom file, is good. I'm lived with it more than year when used SublimeText and can say that is really useful, without variants.  Completed task mostly became uninteresting, so what is the reason to always see it in top of file when i open todo file?  If it feature is conflict with Projects+ Todo+ , is another kind of problem.

**Comment by Cl3r1k** [comment]
"add datetime stamp when task done/canceled"
Sounds interesting) It could be helpfull for me, for example.

**Comment by fabiospampinato** [comment]
> Completed task mostly became uninteresting, so what is the reason to always see it in top of file when i open todo file?

Why not using the `Delete Line` command then?

I'll keep this issue open, I might add the timestamp feature if it gets requested by enough people and/or I'll have time to implement it.

**Comment by EvgeniyMakhmudov** [comment]
> Why not using the Delete Line command then?

The same reason that you marked task "done", instead just delete it - for history.

**Comment by EvgeniyMakhmudov** [comment]
@Cl3r1k if you interesting it, react top message with thumbs up emoji

**Comment by fabiospampinato** [comment]
If the todo list is pretty short I don't see much gain in archiving completed todos, and if the list is long you're probably nesting them, therefor something like this:
```
 ☐ Implement feature A
   ☐ Something to-do
   ✔ Fix issue #123
   ✔ Ensure it's working on foo
 ☐ Implement feature B
```
will look like this when archiving completed todos:
```
 ☐ Implement feature A
   ☐ Something to-do
 ☐ Implement feature B

Archive:
 ✔ Fix issue #123
 ✔ Ensure it's working on foo
```
Now I'm not even sure what those todos are talking about.

------

**But**, I can see some uses for these features when writing todo lists for things other than programming, for instance one might have a list of books to read, and it'll be useful to keep a list of read books and to know when each book was finished.

I'll probably add these features as soon as I get some free time to work on them.

Thanks for the suggestions.

**Comment by ColonelBundy** [comment]
Pretty much would love to see the features in plaintasks to be added.

**Comment by rfoust** [comment]
I'd also like to see the timestamp and archive features added.  I use them daily in the other editors mentioned, but would like to move everything to VS Code.

**Comment by fabiospampinato** [comment]
I'll work on this tomorrow 👍

**Comment by fabiospampinato** [comment]
Timekeeping and Archive functionalities have been implemented in v1.7.0.

Not all of PlainTasks' functionalities have been ported though, but it should be enough for most.

If you feel something in missing please open a new issue.
